tests/cuda_interop/cuda_interop_kernel_func_tests.hpp

```diff
@@ -0,0 +1,452 @@
+  }
+  assert(is_type_correct[0]);
+}
+
+/** check get_native() returns the correct type for a local accessor
+ */
+template <>
+void test_local_target_accessor<c2>(sycl::queue &queue,
+                                    sycl_cts::util::logger &log,
+                                    const std::string &typeName) {
+  size_t constexpr size = 1;
+  bool is_type_correct[size] = {false};
+  {
+    sycl::buffer<bool> is_type_buff(is_type_correct, sycl::range<1>(size));
+
+    queue.submit([&](sycl::handler &cgh) {
+      auto acc =
+          sycl::accessor<c2, 1, sycl::access::mode::write, sycl::target::local>(
+              sycl::range<1>(size), cgh);
+      auto is_type_acc =
+          is_type_buff.get_access<sycl::access::mode::write>(cgh);
+
+      cgh.single_task([=]() {
+        auto native_handle =
+            sycl::get_native<sycl::backend::cuda>(acc);
+        is_type_acc[0] = std::is_same_v<decltype(native_handle), c2 *>;
+      });
+    });
+  }
+
+  if (!is_type_correct[0]) {
+    log.note(
+        "Test for CUDA kernel function local accessor interop failed for \"" +
+        typeName + "\" type");
+  }
+  assert(is_type_correct[0]);
+}
+
+/** @brief
+ *  @tparam
+ *  @tparam
+ */
+template <typename T>
+struct run_all_tests {
+  void operator()(sycl::queue &queue, sycl_cts::util::logger &log,
+                  const std::string &typeName) {
+    // Check type for accessor
+    test_accessor<T>(queue, log, typeName);
+    // Check vector types for accessor
+    test_accessor<sycl::vec<T, 1>>(queue, log, typeName);
+    test_accessor<sycl::vec<T, 2>>(queue, log, typeName);
+    test_accessor<sycl::vec<T, 3>>(queue, log, typeName);
+    test_accessor<sycl::vec<T, 4>>(queue, log, typeName);
+    test_accessor<sycl::vec<T, 8>>(queue, log, typeName);
+    test_accessor<sycl::vec<T, 16>>(queue, log, typeName);
+    // Check marray for accessor
+    test_accessor<sycl::marray<T, 2>>(queue, log, typeName);
+    test_accessor<sycl::marray<T, 5>>(queue, log, typeName);
+    test_accessor<sycl::marray<T, 10>>(queue, log, typeName);
+
+    // Check type for constant buffer accessor
+    test_constant_buffer_accessor<T>(queue, log, typeName);
+    // Check vector types for constant buffer accessor
+    test_constant_buffer_accessor<sycl::vec<T, 1>>(queue, log, typeName);
+    test_constant_buffer_accessor<sycl::vec<T, 2>>(queue, log, typeName);
+    test_constant_buffer_accessor<sycl::vec<T, 3>>(queue, log, typeName);
+    test_constant_buffer_accessor<sycl::vec<T, 4>>(queue, log, typeName);
+    test_constant_buffer_accessor<sycl::vec<T, 8>>(queue, log, typeName);
+    test_constant_buffer_accessor<sycl::vec<T, 16>>(queue, log, typeName);
+    // Check marray for constant buffer accessor
+    test_constant_buffer_accessor<sycl::marray<T, 2>>(queue, log, typeName);
+    test_constant_buffer_accessor<sycl::marray<T, 5>>(queue, log, typeName);
+    test_constant_buffer_accessor<sycl::marray<T, 10>>(queue, log, typeName);
```

Comment:
You probably can use `for_type_vector_marray` function from `common/type_coverage.h`  to cover `sycl::vec` and `sycl::marray` of these sizes.

```
for_type_vectors_marray<test_accessor,T>(queue, log, typeName);
for_type_vectors_marray<test_constant_buffer_accessor,T>(queue, log, typeName);
```

Reply:
That is a great idea. I have updated the PR to use `for_type_vectors_marray`.